reframe/core/launchers/mpi.py

```diff
@@ -3,15 +3,54 @@
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
+import semver
+import re
+
+import reframe.utility.osext as osext
 from reframe.core.backends import register_launcher
 from reframe.core.launchers import JobLauncher
+from reframe.core.logging import getlogger
 from reframe.utility import seconds_to_hms
 
 
 @register_launcher('srun')
 class SrunLauncher(JobLauncher):
+    def __init__(self):
+        self.options = []
+        self.explicit_cpus_per_task = True
+        try:
+            out = osext.run_command('srun --version')
+            match = re.search('slurm (\d+).0*(\d+).0*(\d+)', out.stdout)
```

Comment:
There is no need to match the 0s. You can then convert them to an int and pass them to `VersionInfo`. Also you must match explicitly the dots.

```suggestion
            match = re.search('slurm (\d+)\.(\d+)\.(\d+)', out.stdout)
```